xmppserver/src/main/java/org/jivesoftware/openfire/group/GroupManager.java

```diff
@@ -789,6 +625,423 @@ private void evictCachedUserSharedGroups() {
+    {
+        final Group updatedGroup;
+        try {
+            // By forcing the lookup, the cache will automatically receive the required refresh.
+            updatedGroup = getGroup(group.getName(), true);
+        } catch (GroupNotFoundException e) {
+            Log.error("Group '{}' was not found after a property (key: '{}') was removed from it. This is indicative of a bug in Openfire. Please consider reporting it.", group.getName(), key, e);
+            return;
+        }
+        propertyChangePostProcess(updatedGroup, key, originalValue);
+
+        // Fire event.
+        final Map<String, Object> params = new HashMap<>();
+        params.put("propertyKey", key);
+        params.put("type", "propertyDeleted");
+        params.put("originalValue", originalValue);
+
+        GroupEventDispatcher.dispatchEvent(updatedGroup, GroupEventDispatcher.EventType.group_modified, params);
+    }
+
+    /**
+     * Updates the caches maintained by this manager and dispatches events that reflect a group having all of its
+     * properties removed.
+     *
+     * This method is intended to be invoked to update the internal state of GroupManager after a group has been
+     * updated. It should rarely be invoked by code that is not part of the GroupManager implementation.
+     *
+     * @param group The group that was modified.
+     * @param originalProperties The properties of the group prior to the removal.
+     */
+    public void propertiesDeletedPostProcess(@Nonnull final Group group, @Nonnull final Map<String, String> originalProperties)
+    {
+        final Group updatedGroup;
+        try {
+            // By forcing the lookup, the cache will automatically receive the required refresh.
+            updatedGroup = getGroup(group.getName(), true);
+        } catch (GroupNotFoundException e) {
+            Log.error("Group '{}' was not found after all its properties were removed from it. This is indicative of a bug in Openfire. Please consider reporting it.", group.getName(), e);
+            return;
+        }
+
+        // Make sure that 'shared roster' changes are processed.
+        for (final Map.Entry<String, String> entry : originalProperties.entrySet()) {
+            GroupManager.getInstance().propertyChangePostProcess(group, entry.getKey(), entry.getValue());
+        }
+
+        // Fire event.
+        final Map<String, Object> event = new HashMap<>();
+        event.put("type", "propertyDeleted");
+        event.put("propertyKey", "*");
+        GroupEventDispatcher.dispatchEvent(updatedGroup, GroupEventDispatcher.EventType.group_modified, event);
+    }
+
+    // A generic method to process the effect of property changes to contact list sharing. Re-used by all property change post-processing methods.
+    private void propertyChangePostProcess(final Group group, final String key, final String originalValue)
+    {
+        switch (key) {
+            case "sharedRoster.showInRoster": {
```

Comment:
Should these magic Strings be constants so that typo's don't creep in?

* "sharedRoster.showInRoster"
* "everybody"
* "sharedRoster.groupList"